### PR TITLE
[3.1] Docs: TreeList OnRowClick and OnRowDoubleClick events

### PR DESCRIPTION
**Pull request by svdimitr**

### CODE REVIEW

File: components/treelist/events.md [hunk 1]
```diff
@@ -946,6 +948,214 @@ Fires when the item selection is enabled and the user changes the selected [item
 }
 ````
 
+### OnRowClick
+
+The `OnRowClick` event fires as a response to the user clicking on a row of the Grid. Clicking on the `TreeListCommandButton`, select row `CheckBox`, expanding a parent row or when the row is in `edit/insert mode` will not trigger the event.
```
Comment (dimodi) [inline components/treelist/events.md:953]:
```suggestion
The `OnRowClick` event fires when the user clicks on a TreeList row. The event does not fire when clicking on:
* `TreeListCommandButton`
* selection `CheckBox`
* expand/collapse button
* row in edit mode
```

File: components/treelist/events.md [hunk 2]
```diff
@@ -946,6 +948,214 @@ Fires when the item selection is enabled and the user changes the selected [item
...
+
+The event handler receives a `TreeListRowClickEventArgs` object which provides the model of the clicked row in the `Item` field that you can cast to your model type.
```
Comment (dimodi) [inline components/treelist/events.md:955]:
```suggestion
The event handler receives a `TreeListRowClickEventArgs` object. It provides the model of the clicked row in the `Item` field that you can cast to your model type.
```

File: components/treelist/events.md [hunk 3]
```diff
@@ -946,6 +948,214 @@ Fires when the item selection is enabled and the user changes the selected [item
...
+        }
+
+        return await Task.FromResult(data);
+    }
+}
+````
+
+### OnRowDoubleClick
+
+The `OnRowDoubleClick` event fires as a response to the user double clicking on a row of the Grid. Clicking on the `GridCommandButton`, select row `CheckBox`, expanding a parent row or when the row is in `edit/insert mode` will not trigger the event.
```
Comment (dimodi) [inline components/treelist/events.md:1057]:
```suggestion
The `OnRowDoubleClick` event fires when the user double-clicks on a TreeList row. The event does not fire when clicking on:
* `TreeListCommandButton`
* row selection checkbox
* expand/collapse button
* row in edit mode
```

File: components/treelist/events.md [hunk 4]
```diff
@@ -946,6 +948,214 @@ Fires when the item selection is enabled and the user changes the selected [item
...
+
+The event handler receives a `TreeListRowClickEventArgs` object which provides the model of the clicked row in the `Item` field that you can cast to your model type.
```
Comment (dimodi) [inline components/treelist/events.md:1059]:
```suggestion
The event handler receives a `TreeListRowClickEventArgs` object. It provides the model of the clicked row in the `Item` field that you can cast to your model type.
```

**Review by dimodi — APPROVED** [review]
For better readability, prefer shorter sentences and lists.

**Comment by jivanova** [comment]
we included the OnContextMenu event as well - could you please add it?

**Pull request merged**